src/main/java/com/example/provider/VehicleServlet.java

```diff
@@ -82,35 +88,46 @@ public VehicleServlet(
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ServletUtils.setStandardResponseHeaders(response);
-
+    
     String vehicleId;
 
-    try {
-      vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
-    } catch (IllegalArgumentException e) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No Vehicle id provided.");
-      return;
-    }
+    vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
 
-    logger.info(String.format("Getting vehicle with vehicleID: %s", vehicleId));
+    if (vehicleId != null && !vehicleId.isEmpty()) {
+      logger.info(String.format("Getting vehicle with vehicleID: %s", vehicleId));
 
-    GetVehicleRequest getVehicleRequest =
+      GetVehicleRequest getVehicleRequest =
         GetVehicleRequest.newBuilder().setName(VehicleUtils.getVehicleName(vehicleId)).build();
 
-    Vehicle vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
-    servletState.setVehicleId(vehicleId);
+      Vehicle vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+      servletState.setVehicleId(vehicleId);
 
-    logger.info(String.format("Vehicle:\n%s", vehicle));
+      logger.info(String.format("Vehicle:\n%s", vehicle));
 
-    if (vehicle != null) {
-      if (tripMatcher.triggerMatching(vehicle, vehicleId)) {
-        vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+      if (vehicle != null) {
+        if (tripMatcher.triggerMatching(vehicle, vehicleId)) {
+          vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+        }
       }
-    }
 
-    response.getWriter().print(GsonProvider.get().toJson(vehicle));
+      response.getWriter().print(GsonProvider.get().toJson(vehicle));
+      response.getWriter().flush();
+      return;
+    } 
+    ListVehiclesRequest listVehiclesRequest = ListVehiclesRequest.newBuilder()
+      .setParent(VehicleUtils.PROVIDER_NAME)
+      .setVehicleState(VehicleState.ONLINE)
+      .setPageSize(100)
+      .build();
+
+    logger.info(String.format("Getting a list of all vehicles"));
```

Comment:
Nit: can probably remove 'String.format' and just inline the string

Reply:
Or is there any other info we think might be useful to log? (ex: page size)

Reply:
Done